### PR TITLE
Gdal shared build

### PR DESCRIPTION
**Pull request by bugbrains**
update to gdal 1.11.2
add openjpeg for JPEG2000 driver
activate armadillo
deactivate jasper (lib is highly inactive and also openjpeg has better support for JPEG200 images)

geotiff without ltmain.sh patch have link error from gdal on 64bit shared

### CODE REVIEW

**Comment by TimothyGu** [comment]
I just merged #699. Are these patches still needed?

**Comment by bugbrains** [comment]
yes jasper is inactive now. and openjpeg is now used in gdal for reading JPEG2000 images. 

https://github.com/bugbrains/mxe/commit/29f8a035095f9cba529c5bfe061e0bb725007715 is needed for armadillo. I cannot build boost with c++11 extension.

finally ltmain.sh commit https://github.com/bugbrains/mxe/commit/55ea9efdcb164e641a16e61af550a1955d6e45b6 is needed for geotiff x86_64. I cannot find any other solution to fix geotiff link errors in GDAL.

**Comment by TimothyGu** [comment]
Then can you please fix the merge conflicts?

**Comment by bugbrains** [comment]
done.

File: src/libgeotiff-1-libtoolfilemagic.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff -bur libgeotiff-1.4.0.orig/ltmain.sh libgeotiff-1.4.0/ltmain.sh
+--- libgeotiff-1.4.0.orig/ltmain.sh	2014-10-24 01:20:03.172985419 +0530
++++ libgeotiff-1.4.0/ltmain.sh	2014-10-24 01:58:50.983491081 +0530
+@@ -2560,7 +2560,7 @@
+     ;;
+   *ar\ archive*) # could be an import, or static
+     if eval $OBJDUMP -f $1 | $SED -e '10q' 2>/dev/null |
+-       $EGREP 'file format pe-i386(.*architecture: i386)?' >/dev/null ; then
++       $EGREP 'file format pe-i386(.*architecture: i386)?|pe-x86-64' >/dev/null ; then
```
Comment (TimothyGu) [inline src/libgeotiff-1-libtoolfilemagic.patch]: The best solution here is running `autoreconf -fi` before configuring.

Also, [libtool upstream](http://git.savannah.gnu.org/cgit/libtool.git/plain/build-aux/ltmain.in) changed it to

```diff
-       $EGREP 'file format pe-i386(.*architecture: i386)?' >/dev/null ; then
+       $EGREP 'file format (pei*-i386(.*architecture: i386)?|pe-arm-wince|pe-x86-64)' >/dev/null; then
```

**Comment by bugbrains** [comment]
I had removed ltmain.sh patch and used autoreconf -fi and confirmed that x86_64 gdal builds fine.

configure.ac needs a fix
https://github.com/bugbrains/mxe/commit/ca2549174c6ae817e30168044315dc9a87de56d5#diff-d1e67500ef97a99e7feabc343978d94c

**Comment by TimothyGu** [comment]
Thanks!

**Comment by bugbrains** [comment]
Thanks to you. more pull request on the way..

**Pull request merged**